test/TestSrt.cpp

```diff
@@ -381,3 +427,135 @@ TEST(TestSrt, SingleSender) {
+        bool successfulWait = connectedCondition.wait_for(lock, std::chrono::seconds(2), [&]() { return connected; });
+        ASSERT_TRUE(successfulWait) << "Timeout waiting for client to connect";
+    }
+
+    size_t nClients = 0;
+    server.getActiveClients([&](std::map<SRTSOCKET, std::shared_ptr<SRTNet::NetworkConnection>>& activeClients) {
+        nClients = activeClients.size();
+        for (const auto& socketNetworkConnectionPair : activeClients) {
+            std::pair<std::string, uint16_t> peerIPAndPort =
+                getPeerIpAndPortFromSRTSocket(socketNetworkConnectionPair.first);
+            EXPECT_EQ(peerIPAndPort.first, "127.0.0.1");
+            EXPECT_EQ(peerIPAndPort.second, 8011);
+
+            std::pair<std::string, uint16_t> ipAndPort =
+                getBindIpAndPortFromSRTSocket(socketNetworkConnectionPair.first);
+            EXPECT_EQ(ipAndPort.first, "127.0.0.1");
+            EXPECT_EQ(ipAndPort.second, 8010);
+        }
+    });
+    EXPECT_EQ(nClients, 1);
+
+    std::pair<std::string, uint16_t> serverIPAndPort = getBindIpAndPortFromSRTSocket(server.getBoundSocket());
+    EXPECT_EQ(serverIPAndPort.first, "127.0.0.1");
+    EXPECT_EQ(serverIPAndPort.second, 8010);
+
+    std::pair<std::string, uint16_t> clientIPAndPort = getBindIpAndPortFromSRTSocket(client.getBoundSocket());
+    EXPECT_EQ(clientIPAndPort.first, "127.0.0.1");
+    EXPECT_EQ(clientIPAndPort.second, 8011);
+}
+
+TEST(TestSrt, AutomaticPortSelection) {
+    SRTNet server;
+    SRTNet client;
+
+    auto serverCtx = std::make_shared<SRTNet::NetworkConnection>();
+    auto clientCtx = std::make_shared<SRTNet::NetworkConnection>();
+    clientCtx->mObject = 42;
+
+    std::condition_variable connectedCondition;
+    std::mutex connectedMutex;
+    bool connected = false;
+
+    // notice when client connects to server
+    server.clientConnected = [&](struct sockaddr& sin, SRTSOCKET newSocket,
+                                 std::shared_ptr<SRTNet::NetworkConnection>& ctx) {
+        {
+            std::lock_guard<std::mutex> lock(connectedMutex);
+            connected = true;
+        }
+        connectedCondition.notify_one();
+        auto connectionCtx = std::make_shared<SRTNet::NetworkConnection>();
+        connectionCtx->mObject = 1111;
+        return connectionCtx;
+    };
+
+    ASSERT_TRUE(server.startServer("0.0.0.0", 0, 16, 1000, 100, SRT_LIVE_MAX_PLSIZE, kValidPsk, false, serverCtx));
+
+    std::pair<std::string, uint16_t> serverIPAndPort = getBindIpAndPortFromSRTSocket(server.getBoundSocket());
+    EXPECT_EQ(serverIPAndPort.first, "0.0.0.0");
+    EXPECT_GT(serverIPAndPort.second, 1024); // We expect it won't pick a privileged port
+
+    // TODO remove me
+    std::cout << "Server selected port " << serverIPAndPort.second << std::endl;
+
+    ASSERT_TRUE(client.startClient("127.0.0.1", serverIPAndPort.second, "0.0.0.0", 0, 16, 1000, 100, clientCtx,
+                                   SRT_LIVE_MAX_PLSIZE, kValidPsk));
```

Comment:
Maybe change this since both startServer and startClient have so many arguments?
```suggestion
    const int kAnyPort = 0;
    ASSERT_TRUE(server.startServer("0.0.0.0", kAnyPort, 16, 1000, 100, SRT_LIVE_MAX_PLSIZE, kValidPsk, false, serverCtx));

    std::pair<std::string, uint16_t> serverIPAndPort = getBindIpAndPortFromSRTSocket(server.getBoundSocket());
    EXPECT_EQ(serverIPAndPort.first, "0.0.0.0");
    EXPECT_GT(serverIPAndPort.second, 1024); // We expect it won't pick a privileged port

    // TODO remove me
    std::cout << "Server selected port " << serverIPAndPort.second << std::endl;

    ASSERT_TRUE(client.startClient("127.0.0.1", serverIPAndPort.second, "0.0.0.0", kAnyPort, 16, 1000, 100, clientCtx,
                                   SRT_LIVE_MAX_PLSIZE, kValidPsk));
```

Reply:
Good suggestion!